_studio/shared/src/libmfx_allocator.cpp

```diff
@@ -41,7 +44,11 @@ mfxStatus mfxDefaultAllocator::AllocBuffer(mfxHDL pthis, mfxU32 nbytes, mfxU16 t
     if(!mid)
         return MFX_ERR_NULL_PTR;
     mfxU32 header_size = ALIGN32(sizeof(BufferStruct));
-    mfxU8 *buffer_ptr=(mfxU8 *)malloc(header_size + nbytes + DEFAULT_ALIGNMENT_SIZE);
+#if defined (MFX_VA_LINUX)
+    mfxU8 *buffer_ptr = (mfxU8 *)aligned_alloc(sysconf(_SC_PAGESIZE), header_size + nbytes + DEFAULT_ALIGNMENT_SIZE);
```

Comment:
Please also add a comment in code why we don't do the same alignment for windows

Reply:
Done.

Reply:
Ditto in samples change. Windows has _algined_malloc and _aligned_free pair. Introduce a wrapping macro which uses appropriate version or use std as @mgonchar suggested.

Reply:
@dvrogozh, yes, you are right. But align_alloc also std function, but currently not implemented in windows.  I will try to investigate both ways which you suggest for choose the best one.

Reply:
@AntonGrishin unfortunately ```std::aligned_alloc``` is part of C++17 which is not supported in MSDK project. Otherwise it would have solved our problem. ```_algined_malloc``` which is suggested by @dvrogozh is properly supported by MSVC, see [here ](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/aligned-malloc?view=vs-2019)